src/core/toga/window.py

```diff
@@ -270,27 +270,34 @@ def open_file_dialog(self, title, initial_directory=None, file_types=None, multi
         """ This opens a native dialog where the user can select the file to open.
         It is possible to set the initial folder and only show files with specified file extensions.
         If no path is returned (eg. dialog is canceled), a ValueError is raised.
+
         Args:
-            title (str): The title of the dialog window.
-            initial_directory(str): Initial folder displayed in the dialog.
-            file_types: A list of strings with the allowed file extensions.
+            title (str): The title of the dialog window (ignored on Android)
+            initial_directory(str): Initial folder displayed in the dialog. On Android, this needs to be a content URI,
+                e.g. 'content://com.android.externalstorage.documents/document/primary%3ADownload%2FTest-dir'
+            file_types: A list of strings with the allowed file extensions. On Android, these must be MIME types,
```

Comment:
Platform specific instructions like these aren't workable in a cross-platform toolkit. We need to find a common format that will work for all platforms, even if some data won't be used on some platforms.

Reply:
Why I made a specific example? Because the methods do_open_file and do_open_folder need to be async for Android. 
Will this not make problems when running on other platforms?
Besides: the example might not fit on the Android screen - and the ScrollContainer implementation for Android is not yet there.
You can remove the filebrowser example if you want - I only needed it for testing the PR

Reply:
This is strange: I can't make the dialogs example work, even when I add `async` to action_open_file_dialog (which makes it unusable under Windows):
The file chooser **does** open, but I don't get the chosen file back.
```
    async def action_open_file_dialog(self, widget):
        try:
            if toga.platform.current_platform == 'android':
                print('Open file dialog for Android')
                fname = await self.app.main_window.open_file_dialog(
                    title="Choose a file",
                    multiselect=False)
                print('Filename:' + str(fname))
            else:
                fname = self.main_window.open_file_dialog(
                    title="Open file with Toga",
                    multiselect=False
                )
            if fname is not None:
                self.label.text = "File to open:" + str(fname)
            else:
                self.label.text = "No file selected!"
        except ValueError:
            self.label.text = "Open file dialog was canceled"
```
I see the first print before the await line, but not the second after the await line. Very strange!

Reply:
I made some tests with the dialogs example. I can it work on Android and Windows by doing this:
```
        if toga.platform.current_platform == 'android':
            btn_open = toga.Button('Open File', on_press=self.action_open_file_dialog_android, style=btn_style)
        else:
            btn_open = toga.Button('Open File', on_press=self.action_open_file_dialog, style=btn_style)
```
and then defining action_open_file_dialog_android as async

So, yes, I could probably do without the filebrowser example

Reply:
OK, I now extended the dialogs example so it also works for Android.
So, I will remove the filebrowser example in the next push.